packages/url_launcher/url_launcher_android/example/integration_test/url_launcher_test.dart

```diff
@@ -20,4 +20,27 @@ void main() {
     // sms:, tel:, and mailto: links may not be openable on every device, so
     // aren't tested here.
   });
+
+  testWidgets('launch and close', (WidgetTester _) async {
+    final UrlLauncherPlatform launcher = UrlLauncherPlatform.instance;
+
+    // Launch a url then close.
+    expect(
+        await launcher.launch('https://flutter.dev',
+            useSafariVC: true,
+            useWebView: true,
+            enableJavaScript: false,
+            enableDomStorage: false,
+            universalLinksOnly: false,
+            headers: <String, String>{'my_header_key': 'my_header_value'}),
+        true);
+    // Allow time for page to load.
+    await Future.delayed(Duration(seconds: 3));
```

Comment:
@bparrishMines @stuartmorgan open to suggestions for how I can make this less hacky. For example some flutter event I can wait on during a normal shutdown. If I modify the android code to cause a crash and I dont await here the tests pass when they should not.